### PR TITLE
Fix provider vs current APIs

### PR DESCRIPTION
**Pull request by chicco785**
@arslanbekov is a working in progress based on changes to the client.

happy to have a chat around this

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,4 +1,4 @@
-OPENVPN_HOST=
+OVPN_HOST=
```
Comment (chicco785) [inline .env.example:1]: to align with https://github.com/OpenVPN/terraform-provider-openvpn-cloud/blob/main/e2e/integration_test.go#L16

File: client/client.go [hunk 2]
```diff
@@ -1,92 +0,0 @@
-package client
```
Comment (chicco785) [inline client/client.go:1]: i removed it, I am not sure this is useful at all, given that it's not different from the actual client imported.

File: client/client_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package client
```
Comment (chicco785) [inline client/client_test.go:1]: may it is better to have something like this in https://github.com/OpenVPN/cloudconnexa-go-client

File: cloudconnexa/data_source_connector.go [hunk 4]
```diff
@@ -45,6 +45,11 @@ func dataSourceConnector() *schema.Resource {
 				Computed:    true,
 				Description: "The IPV6 address of the connector.",
 			},
+			"profile": {
```
Comment (chicco785) [inline cloudconnexa/data_source_connector.go:48]: so now we can retrieve also ovpn profiles

File: cloudconnexa/resource_connector.go [hunk 5]
```diff
@@ -54,6 +55,11 @@ func resourceConnector() *schema.Resource {
 				Computed:    true,
 				Description: "The IPV6 address of the connector.",
 			},
+			"profile": {
```
Comment (chicco785) [inline cloudconnexa/resource_connector.go:58]: so now we can retrieve also ovpn profiles

File: cloudconnexa/resource_network.go [hunk 6]
```diff
@@ -73,7 +74,7 @@ func resourceNetwork() *schema.Resource {
 							Default:     "Managed by Terraform.",
 							Description: "The default route description.",
 						},
-						"value": {
+						"subnet": {
```
Comment (chicco785) [inline cloudconnexa/resource_network.go:77]: this is required to populate correctly this field

File: cloudconnexa/resource_network.go [hunk 7]
```diff
@@ -134,6 +135,11 @@ func resourceNetwork() *schema.Resource {
 							Computed:    true,
 							Description: "The IPV6 address of the default connector.",
 						},
+						"profile": {
+							Type:        schema.TypeString,
```
Comment (chicco785) [inline cloudconnexa/resource_network.go:139]: we retrieve also the profile

File: example/backend.tf [hunk 8]
```diff
@@ -2,8 +2,8 @@ terraform {
   backend "local" {}
   required_providers {
     openvpncloud = {
-      source  = "OpenVPN/openvpn-cloud"
-      version = "0.0.11"
+      source  = "OpenVPN/cloudconnexa"
```
Comment (chicco785) [inline example/backend.tf:5]: given that it seems that you want to target cloudconnexa as name

**Comment by chicco785** [comment]
changes to the provider have been tested for few use cases, most probably there are other fixes needed, e.g.:
- updating a network by changing the route is not working correctly;
- destroy a ip_service (if it's the only one attached to a network) is not working (but this is more a logic problem at the api level), despite you destroy also the connected network (maybe playing with lifecycles will make it work, not sure).

**Comment by arslanbekov** [comment]
@chicco785 Thank you for this. Very valuable!

**Pull request merged**